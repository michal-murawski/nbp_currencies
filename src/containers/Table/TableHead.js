import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import {
  TableHead as TableHeadMUI,
  TableRow,
  TableCell,
  TableSortLabel,
} from 'material-ui/Table';
import { TableHeadStyles } from './styles';

class TableHead extends Component {
  constructor() {
    super();

    this.state = {
      activeIndex: null,
    };
  }

  handleOnClick = (sorter, index) => () => {
    if (sorter) {
      this.setState({ activeIndex: index });
      this.props.onHeadCellClick(sorter);
    }
  };

  render() {
    const { classes, headerLabels, direction } = this.props;
    const { activeIndex } = this.state;

    return (
      <TableHeadMUI>
        <TableRow classes={{ root: classes.tableHeadRow }}>
          {headerLabels.map((label, index) => (
            <TableCell
              onClick={this.handleOnClick(label.sorter, index)}
              key={label.label}
              classes={{ head: classes.title }}
            >
              {label.sorter ? (
                <TableSortLabel direction={direction} active={activeIndex === index}>
                  {label.label}
                </TableSortLabel>
              ) : (
                label.label
              )}
            </TableCell>
          ))}
        </TableRow>
      </TableHeadMUI>
    );
  }
}

TableHead.propTypes = {
  onHeadCellClick: PropTypes.func,
  direction: PropTypes.oneOf(['asc', 'desc']),
  headerLabels: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      sorter: PropTypes.string,
    })
  ),
};

export default withStyles(TableHeadStyles)(TableHead);
