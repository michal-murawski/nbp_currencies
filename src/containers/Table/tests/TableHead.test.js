import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { tableHeadLabels } from 'tests/testData';
import TableHead from '../TableHead';

const mockOnHeadClick = jest.fn();

describe('containers/Table/TableHead', () => {
  it('should render proper number of cells', () => {
    const wrapper = shallow(
      <TableHead
        headerLabels={tableHeadLabels}
        onHeadCellClick={mockOnHeadClick}
        direction="asc"
      />
    );

    expect(toJson(wrapper.render())).toMatchSnapshot();
  });
});
