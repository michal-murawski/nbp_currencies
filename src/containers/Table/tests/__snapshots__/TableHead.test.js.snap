// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Table/TableHead should render proper number of cells 1`] = `
<thead
  class="MuiTableHead-root-2"
>
  <tr
    class="MuiTableRow-root-3 MuiTableRow-head-4"
  >
    <th
      class="MuiTableCell-root-8 MuiTableCell-head-9 TableHead-title-1"
      scope="col"
    >
      <span
        class="MuiButtonBase-root-21 MuiTableSortLabel-root-16"
        role="button"
        tabindex="0"
      >
        Name
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-24 MuiTableSortLabel-icon-18 MuiTableSortLabel-iconDirectionAsc-20"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          >
            
          </path>
        </svg>
      </span>
    </th>
    <th
      class="MuiTableCell-root-8 MuiTableCell-head-9 TableHead-title-1"
      scope="col"
    >
      <span
        class="MuiButtonBase-root-21 MuiTableSortLabel-root-16"
        role="button"
        tabindex="0"
      >
        Code
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-24 MuiTableSortLabel-icon-18 MuiTableSortLabel-iconDirectionAsc-20"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          >
            
          </path>
        </svg>
      </span>
    </th>
    <th
      class="MuiTableCell-root-8 MuiTableCell-head-9 TableHead-title-1"
      scope="col"
    >
      <span
        class="MuiButtonBase-root-21 MuiTableSortLabel-root-16"
        role="button"
        tabindex="0"
      >
        Value [PLN]
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-24 MuiTableSortLabel-icon-18 MuiTableSortLabel-iconDirectionAsc-20"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          >
            
          </path>
        </svg>
      </span>
    </th>
    <th
      class="MuiTableCell-root-8 MuiTableCell-head-9 TableHead-title-1"
      scope="col"
    >
      Love it!
    </th>
  </tr>
</thead>
`;
